### ISSUE TITLE
Version 2 Frontend Design

### ISSUE THREAD

**Issue by bbertucc** [bbertucc/equalify#492]
## Overview
This ticket includes all aspects of the version 2 design. 

## Requirements
The design should include all features mentioned in #373 and of course be WCAG 2.2 AA conformant. Moreover, the design must be screen reader first. Sighted user second. That means the first drafts should be an outline of what is included in the DOM, read by screen readers. Further drafts can tune the user experience (UX) for sighted users. 

## Outcomes
- [x] **[Screen Reader Spec](https://docs.google.com/document/d/1R7043tj8FHjmxkOJKGKoQw4dKdoFD2Hgb73lMWQaWPc/edit?usp=sharing)**:  includes the different pages and page content, legible for screen reader users and designers. Validated by core team and accessibility pros.
- [x] **[System Architecture Document](https://docs.google.com/document/d/1yp_fVde4USuzvp6Wwb9oFm2LOp3Ej85Z19QfCf7UhG8/edit?tab=t.0)**: System functionality and screen information, helping us optimize performance without compromising features or accessibility. Validated by core team and accessibility pros.
- [x] **[Sales Strategy](https://docs.google.com/document/d/195G3lwmAI2IJduC6xxsCy2v5Nswh9fgzuvKQkV7yKYw/edit?tab=t.0#heading=h.v91tnzn22hfy)**: Clear strategy for selling to and onboarding Higher Ed organizations.

## Steps to Close This Issue
- [x] Develop Screen Reader Spec, System Architecture, and Sales Strategy doc from #373 - @bbertucc will incorporate a spec with all featured described below (draft by 12/5 meeting, finalized by 12/9 meeting)
- [x] #512
- [x] Develop schema of existing DB #507 that can be incorporated into the System architecture
- [x] Update Screen Reader Spec, System Architecture, and Sales Strategy - completed by @bbertucc after the previous process item
- [x] @bbertucc Validates Screen Reader Spec with Lucy, Alex, Raga, and other blind accessibility professionals who show interest - each offered $70 for their input.
- [x] Update docs as needed - budgeted by @bbertucc after the previous process item
- [x] Final Validation of Screen Reader Spec and System Architecture with core contributors (@wilsuriel03 @alexstine @azdak @heythisischris @kevinandrews1 and @BaronWolfenstein) team in an hour meeting led by @bbertucc - each person is paid $70.

## Features
The following features were defined through user study work (#501) and should be included in initial design documents. These features will be ticked off as they are added to the docs:
- [x] **Use "URLs" Instead of "Properties"**: Users should import URLs versus adding properties. Properties are an arbitrary abstract and complicate the way we handle events like a URL being deleted from a sitemap.
- [x] **Notification Feed and Settings**: Users need to be able to decide when they want to be notified and how (in-app or email). Notification could be a daily/monthly digest and any critical issues (like a property URL is unreachable).
- [ ] **Drupal and WordPress Integration**: That shows accessibility across many properties, in addition to giving authors and editors feedback of accessibility issues after save.
- [x] **Simple Reporting Statistic**: A simple reporting statistic on the dashboard that gives a quick picture of the success of URLS in reports (example: "F" grade)
- [x] **Option to Ignore**: we need the option to ignore one or many messages/nodes. When a user ignores, they must write a message of why they are ignoring. 
- [x] **Issue Density on Main Report Chart**: Users are worried when a lot of new issues are displayed on the chart, as noted in #317. For that reason, we should display the issue density rather than number of issues. Issue density can be found by dividing total issues by total pages. That number can be logged over time.
- [x] **Added Team Members**: Future versions of Equalify will need the ability to add team members and set permission.
- [x] **Shareable Reports**: Users have requested shareable reports. That would mean adding sharing settings that allow you to make a link public.
- [x] **"Create Issue" Feature and PM Integrations:** We've started to build this feature with #365. I imagine that this could become a much more robust feature, where you can choose to accept what AI generated before you post it to GitHub or Gira.
- [x]   **Scan Customizer**: To compete with Enterprise solutions, we'll need to offer a scan customizer. That would allow users to turn on and off different scans (axe-core, Editora11y, WAVE, spellchecker, broken links, readability). Users would also be able to write custom scan rules, like creating a "Needs Review" alert when the words "Accessibility Statement" are found.
- [x] **Add Log to Report Page**: Report page should have a log of when reports were generated and when ignored.
- [x] **Page History**: @azdak is building out an automatic monitoring system. When looking at a detail view of a page, we'll need to show the history of it being automatically scanned through our monitoring system. Here's the type of data we're working with: https://monitor-dashboard.equalify.app/monitor/667ed3c76e2226011bba2e8c
- [x] **Node(s) History**: The users will need to see an audit trail of what happened with individual or a group of nodes. For instance, if a node was ignored or equalified. They should see the time of the node update and a message related to it. 
- [ ] **Error Highlighting**: The error should be highlighted on a page, so when you're looking at the page you can see where the specific issue is. That can also be achieved with screenshots. 
- [x] **In-App Purchasing**: Admins should be able to purchase features and user accounts from inside the app.
- [x] **Display filters of Individual Report on Reports Page**: People identify reports by filters. We should probably add badges of the filters on the report page so folks know what the report is about.
- [x] **Moving Scan Logs into Properties**: Scans are related to properties, so they should be moved onto the page.
- [x] **Add Scan Frequency to Properties**: People need to choose scan frequency (Manually, Daily, Weekly, Monthly, or On Page Update)

**Comment by kevinandrews1** [comment]
@bbertucc Modify "compliant" to "conformant".

**Comment by bbertucc** [comment]
@azdak does this make sense to you? Trying to outline the design work that goes into our next iteration so we don't get stuck in similar problems we had before. I hope we can do this design phase before January so we can more accurately estimate dev work.

**Comment by azdak** [comment]
@bbertucc yes- a couple months ago I mentioned in the slack that I thought there might be a way to _truly_ do accessibility-first UX in a way that would both produce an accessible interface, but also significantly improve overall sighted usability as well, and I think (re-)starting our overall redesign process with a **text-based functional spec** could really move us in that direction, as it will force us to really think about the information architecture, ordering of elements, and user inputs/outputs of each individual screen. 

So I think we should do item 1 on this list, but really make sure we're really devoting thought/time to it- imo we should list every individual screen on the app, and then work through to each component and input. I'd actually suggest doing a **requirements spec** first, to make the big structural decisions, like:

```
Reports Section
	Functionality: 
		- List all reports
		- Create new report
		- Delete reports
		- [Anything else?]
	Screens:
		- Default
			- List of reports 
			        - For Each: 
				- [Button to Single Report] View Report
				- [Button to Single Report] Delete Report
				- [Do we need any other info, like number of pages in a report?]
			- Button to Create Report
			- [Do we want or need some type of summary info here?
		- Single Report
			- [...]
		- Create Report [ Should this be a page, or a modal? ]
			- Inputs:
				- [Text Input] Report Name
				- [...]
```
and then after we've hashed the actual functionality out, move to a **screen reader spec**, something like:
```
(Announced items in **stars**)

**Properties Page**
**Main Menu**:
	- **View Reports** [ Link to /reports ]
	- **View Properties** [ Link to /properties ]
	- **View My Account**
- **Add Report** [ Button ]
**All Reports**
        - **Add Report** [ Button ]
        - **Reports List**
	     - **My Report**
	     - [...]
	
```
I do think this would all take some effort and thought, but I think would really help us address some of the overarching problems with information architecture and organization of the app we've had, and do a lot to give us a firm footing for the next six months or year of design/development-

**Comment by bbertucc** [comment]
@azdak @heythisischris i added my budget for the first draft here (5 hours). any comments? can i proceed?

**Comment by BaronWolfenstein** [comment]
I'm @BaronWolfenstein

**Comment by bbertucc** [comment]
A little update: Still through the initial. Plan to have stuff ready by 12/11 to validate with the team.

**Comment by bbertucc** [comment]
Screen reader spec draft 1 now finished: https://docs.google.com/document/d/1R7043tj8FHjmxkOJKGKoQw4dKdoFD2Hgb73lMWQaWPc/edit?usp=sharing

Notified team to schedule next meeting.

**Comment by bbertucc** [comment]
Note: I updated the screen reader spec to HTML, here: https://html-preview.github.io/?url=https://raw.githubusercontent.com/EqualifyEverything/equalify-dashboard/v2/v2/dashboard.html

**Comment by alexstine** [comment]
I really enjoy this spec @bbertucc . Things are much clearer now.

**Comment by kevinandrews1** [comment]
@bbertucc let me know when you're ready for me to review this. Also let me know what kind of feedback you would like and if here is the best place to provide. Thanks.
